tvb/analyzers/fragility.py

```diff
@@ -0,0 +1,172 @@
+        required=True,
+        default=125,
+        doc="Size of the windows to use in milliseconds.")
+
+    samplerate = basic.Float(
+        label="Sample rate of signal in Hz.",
+        required=True,
+        default=1000,
+        doc="Sample rate of the signal in Hz")
+
+    # whether or not to use multiprocessing
+    parallelize = False
+    numcores = psutil.cpu_count() / 2.
+
+    def evaluate(self, linefreq=60):
+        rawdata = self.timeseries.data
+
+        # first filter line noise
+        noisemodel = FilterLinearNoise(samplerate=self.samplerate)
+        rawdata = noisemodel.notchlinenoise(rawdata, linefreq=linefreq)
+
+        # instantiate a mvarmodel object
+        if self.parallelize:
+            mvarmodel = ParallelMvar(self.winsize, self.stepsize, self.samplerate, numcores=numcores)
+            
+            # initialize model
+            mvarmodel.runmvar(rawdata)
+            adjmats = mvarmodel.mergemvarresults()
+        else:
+            # run serial mvar model computationa
+            mvarmodel = SingleMvar(self.winsize, self.stepsize, self.samplerate)
+            adjmats = mvarmodel.runmvar(rawdata, normalize=True)
+        self.adjmats = adjmats
+        return adjmats
+
+    def result_shape(self):
+        "Returns the resulting model."
+        n = self.adjmats.shape 
+        return n
+    def result_size(self):
+        "Returns the storage size in bytes of the mixing matrix of the ICA analysis, assuming 64-bit float."
+        return numpy.prod(self.result_shape()) * 8
+    def _find_summary_info(self):
+        """
+        To be implemented in every subclass.
+        """
+        return None
+
+class fragilitymodel(core.Type):
+    adjmats = arrays.FloatArray(
+        label="Time Series",
+        required=True,
+        doc="The timeseries to which the fragility analysis is to be applied.")
+
+    radius = basic.Float(
+        label="Radius of perturbation.",
+        required=True,
+        default=1000,
+        doc="Radius of perturbation.")
+
+    # whether or not to use multiprocessing
+    parallelize = False
+    numcores = psutil.cpu_count() / 2.
+
+    def _compute_fragilitymetric(self, minnormpertmat):
+        # get dimensions of the pert matrix
+        N, T = minnormpertmat.shape
+        # assert N < T
+        fragilitymat = numpy.zeros((N, T))
+        for icol in range(T):
+            fragilitymat[:,icol] = (numpy.max(minnormpertmat[:,icol]) - minnormpertmat[:,icol]) /\
+                                    numpy.max(minnormpertmat[:, icol])
+        return fragilitymat
+
+    def evaluate(self):
+        adjmats = self.adjmats
+        # instantiate a mvarmodel object
+        if self.parallelize:
+            # initialize model
+            pertmodel = ParallelPert(radius=1.5, perturbtype='C', numcores=numcores)
+            
+            # create temp dir to save files
+            tempdir = None
+            assert tempdir is str
```

Comment:
there's a problem here: `tempdir` will be `None` not `str`

Reply:
Fixed assertion statement.